### PR TITLE
feat: Deep connector Smartlead (Read)

### LINKED ISSUE #911: Add Smartlead deep connector

### PR DESCRIPTION
**Pull request by Cobalt0s**
Closes #911 

# Description
There is no pagination.

# Testing
image

### CODE REVIEW

**Comment by Cobalt0s** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/901?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#902** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/902?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#901** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/901?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#900** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/900?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#899** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/899?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#898** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/898?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Cobalt0s and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: smartlead/connector.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	conn.Client.HTTPClient.ErrorHandler = interpreter.ErrorHandler{
+		JSON: interpreter.NewFaultyResponder(errorFormats, nil),
+		HTML: &interpreter.DirectFaultyResponder{Callback: conn.interpretHTMLError},
+	}.Handle
+
+	return conn, nil
+}
+
+func (c *Connector) Provider() providers.Provider {
+	return providers.Smartlead // TODO check such provider exists in catalog
```
Comment (RajatPawar) [inline smartlead/connector.go:54]: This comment can be removed

**Review by RajatPawar — APPROVED** [review]
Good to go!

File: smartlead/connector.go (hunk 1 above)
Comment (Cobalt0s) [reply smartlead/connector.go:54]: Thats from generator code. My bad

**Pull request merged**